src/pattern-library/components/patterns/data/Icons.js

```diff
@@ -0,0 +1,110 @@
+          replace the single legacy <code>Icon</code> component.
+        </p>
+        <Library.Example title="Migrating to this component">
+          <Next.Changelog>
+            <Next.ChangelogItem status="changed">
+              It is no longer necessary to import SVG source nor registration
+              functions from this package to use icons.
+            </Next.ChangelogItem>
+          </Next.Changelog>
+        </Library.Example>
+      </Library.Pattern>
+      <Library.Pattern title="Usage">
+        <Next.Usage componentName="CancelIcon" />
+        <Library.Example>
+          <Library.Demo withSource>
+            <Icons.CancelIcon />
+          </Library.Demo>
+        </Library.Example>
+      </Library.Pattern>
+
+      <Library.Pattern title="Props">
+        <p>
+          Icon components accept any{' '}
+          <code>
+            <Link href="https://developer.mozilla.org/en-US/docs/Web/API/SVGSVGElement">
+              SVGSVGElement
+            </Link>
+          </code>{' '}
+          attributes as props. Use <code>className</code> (not{' '}
+          <code>classes</code>) to style icons.
+        </p>
+        <Library.Example>
+          <Library.Demo title="Styled icon component" withSource>
+            <Icons.CautionIcon className="text-yellow-notice w-16 h-16" />
+          </Library.Demo>
+        </Library.Example>
+      </Library.Pattern>
+
+      <Library.Pattern title="Sizing icons">
+        <p>
+          Icons are sized at <code>16×16</code> unless styled otherwise.
+        </p>
+        <p>
+          Icons are often sized relative to the surrounding text. The{' '}
+          <code>em</code> spacing value provided by this {"package's"} tailwind
+          preset allows sizing at <code>1em</code>.
+        </p>
+
+        <p>
+          <Link
+            href="https://tailwindcss.com/docs/customizing-spacing"
+            underline="always"
+          >
+            <div className="flex items-center gap-x-1">
+              More information about {"tailwind's"} sizing utility classes
+              <Icons.ExternalIcon />
+            </div>
+          </Link>
+        </p>
+        <Library.Example>
+          <Library.Demo title="Sized icon components" withSource>
+            <div className="space-y-2">
+              <div className="text-xs items-center flex gap-x-2">
+                <Icons.ArrowRightIcon className="w-em h-em" />
+                Icon sized relative to some small text
+              </div>
+              <div className="text-lg items-center flex gap-x-2">
+                <Icons.ArrowRightIcon className="w-em h-em" />
+                Icon sized relative to some larger text
+              </div>
+            </div>
+          </Library.Demo>
+        </Library.Example>
+      </Library.Pattern>
+
+      <Library.Pattern title="Available icon components">
+        <div className="grid grid-cols-4 gap-6">
+          {Object.keys(Icons).map(iconName => {
```

Comment:
You could also use `Object.entries(Icons).map(([iconName, IconComponent]) =>  ...` here.